### ISSUE TITLE
Input doesn't recognize 31st

### ISSUE THREAD

**Issue by william-lima-p** [stla/DateTimeRangePicker#1]
The input won't accept any 'YYYY-MM-31' date, modifying itself to  'YYYY-MM-01' in the server.

![image](https://github.com/stla/DateTimeRangePicker/assets/98840083/b13667dd-364d-4aa3-a7bf-d999615f5b76)

I'm using the shinyapp provided in the documentation:

```
library(DateTimeRangePicker)
library(shiny)

ui <- fluidPage(
  br(),
  sidebarLayout(
    sidebarPanel(
      width = 5,
      tags$fieldset(
        tags$legend("Click to change time"),
        DateTimeRangePickerInput(
          "dtrpicker",
          style = paste0(
            "background-color: chartreuse; ",
            "box-shadow: 0 30px 40px 0 rgba(16, 36, 94, 0.2);"
          )
        )
      )
    ),
    mainPanel(
      width = 7,
      verbatimTextOutput("dtrpicker")
    )
  )
)

server <- function(input, output){
  output[["dtrpicker"]] <- renderPrint({
    input[["dtrpicker"]]
  })
}

if(interactive()){
  shinyApp(ui, server)
}
```

**Comment by stla** [comment]
Thanks! I think I understand why. That's because in R, when I construct the date, I set the day before setting the month. And then, if there's no 31st in the current month, the day is set to the 1st. If you first pick a day in October, say the 16th, and then you pick the 31st, then it works. I will fix this issue, thanks again for the report.

**Comment by stla** [comment]
Fixed in v1.1.0.
